addons/mail/models/mail_thread.py

```diff
@@ -1817,9 +1817,13 @@ def message_post(self, *,
             if subtype_xmlid in ('mail.mt_note', 'mail.mt_comment'):
                 subtype_id, internal = self.env['ir.model.data'].xmlid_to_res_id(subtype_xmlid), subtype_xmlid == 'mail.mt_note'
             else:
-                subtype_id, internal = self.env['mail.message.subtype']._subtype_ref(subtype_xmlid, raise_if_not_found=False)
-        if not subtype_id:
-            subtype_id, internal = self.env['mail.message.subtype']._subtype_ref('mail.mt_note')
+                subtype_id, internal = self.env['mail.message.subtype'].sudo()._subtype_ref(subtype_xmlid, raise_if_not_found=True)
+        elif not subtype_id:
+            subtype_id, internal = self.env['mail.message.subtype'].sudo()._subtype_ref('mail.mt_note', raise_if_not_found=True)
+        else:
+            if 'internal' in kwargs:
+                internal = kwargs['internal']
```

Comment:
We allow internal override from kwargs only on the case "no subtype_xmlid but subtype_id given"?

Reply:
Mmmh, I rewrote a bit that part of code, it was indeed a bit incorrect.